presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java

```diff
@@ -340,12 +342,14 @@ public static long parseHiveTimestamp(String value, DateTimeZone timeZone)
 
     public static boolean isSplittable(InputFormat<?, ?> inputFormat, FileSystem fileSystem, Path path)
     {
-        // ORC uses a custom InputFormat but is always splittable
-        if (inputFormat.getClass().getSimpleName().equals("OrcInputFormat")) {
+        //This check covers, ORC/DWRF/AVRO/RCBINARY/RCTEXT/SEQUENCEFILE/PAGEFILE storage formats.
+        if (inputFormat instanceof FileInputFormat ||
```

Comment:
Seems `EncryptedOrcInputFormat` is the only inheritance of `OrcInputFormat`. So,
```java
inputFormat.getClass().getSimpleName().equals("OrcInputFormat")
```
is equivalent to
```java
inputFormat instance of OrcInputFormat && !(inputFormat instance of EncryptedOrcInputFormat))
```
right?

For `EncryptedOrcInputFormat`, given we are not using the open-source encryption, so we can ignore that and just use `inputFormat instance of OrcInputFormat`.

Reply:
I am not entirely sure what exactly you are trying to infer.  Yes, following is correct.

inputFormat.getClass().getSimpleName().equals("OrcInputFormat")
is equivalent to
inputFormat instance of OrcInputFormat && !(inputFormat instance of EncryptedOrcInputFormat))

This check will ensure isSpittable to true for EncryptedOrcInputFormat.

Reply:
Actually now I realize this change is bad. Its blindly assuming all extensions of FileInputFormat to be splittable, which may not be the case.

Reply:
I am only going to extend the condition for RCFileInputFormat only.